src/constraint/encoding.ts

```diff
@@ -140,6 +140,34 @@ export const ENCODING_CONSTRAINTS: EncodingConstraintModel[] = [
       return true;
     }
   },{
+    name: 'scalePropertiesSupportedByScaleType',
+    description: 'Scale properties must be supported by correct scale type',
+    properties: [Property.SCALE, Property.SCALE_TYPE, Property.TYPE],
+    allowEnumSpecForProperties: true,
+    strict: true,
+    satisfy: (encQ: EncodingQuery, schema: Schema, opt: QueryConfig) => {
+      if (encQ.scale) {
+        const scale: ScaleQuery = encQ.scale as ScaleQuery;
+        const hasScaleType: boolean = (scale.type !== undefined);
```

Comment:
Here `allowEnumSpecForProperties` is `true`.   

If you add a test where `encQ.scale = false` I think you will get error that can't get `type` of undefined.